test/javascript/components/dropdown.js

```diff
@@ -37,6 +44,54 @@ describe('Dropdown', function() {
       sinon.assert.called(spy);
       Foundation.Keyboard.trapFocus.restore();
     });
+
+    it('should open dropdown on button click', function() {
+      $dropdownController = $(getDropdownController()).appendTo('body');
+      $dropdownContainer = $(getDropdownContainer()).appendTo('body');
+      plugin = new Foundation.Dropdown($dropdownContainer, {});
+      plugin.open();
+
+      $dropdownController.on('show.zf.dropdown', function() {
+        $('#my-dropdown').should.not.be.hidden;
+        $('#my-dropdown').should.have.attr('aria-hidden', 'false');
+        $('#my-dropdown').should.have.class('is-open');
+        done();
+      });
+      plugin.close();
+    });
+
+    it('should open dropdown on hover', function() {
+      $dropdownController = $(getDropdownController()).appendTo('body');
+      $dropdownContainer = $(getHoverDropdownContainer()).appendTo('body');
+      plugin = new Foundation.Dropdown($dropdownContainer, {});
+      plugin.open();
+
+      $dropdownController.on('show.zf.dropdown', function() {
+        $('#my-dropdown').should.not.be.hidden;
+        $('#my-dropdown').should.have.attr('aria-hidden', 'false');
+        $('#my-dropdown').should.have.class('is-open');
+        done();
+      });
+      plugin.close();
+    });
+
+    it('should autofocus input on open', function() {
+      $dropdownController = $(getDropdownController()).appendTo('body');
+      $dropdownContainer = $(getAutoFocusDropdownContainer()).appendTo('body');
+      plugin = new Foundation.Dropdown($dropdownContainer, {});
+      plugin.open();
+
+      $focusedElement = '<input type="text" placeholder="should auto focus" id="focus-text" />';
```

Comment:
We can also put an ID on the autofocus (that we pass as parameter) and test it on the `activeElement`. I find that more enduring.